pyteal/ir/tealcomponent.py

```diff
@@ -1,14 +1,23 @@
 from abc import ABC, abstractmethod
-from typing import List, Optional, TYPE_CHECKING
 from contextlib import AbstractContextManager
+from typing import TYPE_CHECKING, List, cast
+
+from pyteal.stack_frame import NatalStackFrame
 
 if TYPE_CHECKING:
     from pyteal.ast import Expr, ScratchSlot, SubroutineDefinition
 
 
 class TealComponent(ABC):
-    def __init__(self, expr: Optional["Expr"]):
-        self.expr = expr
+    def __init__(self, expr: "Expr | None"):
+        self.expr: Expr | None = expr
+
+        # ALL BELOW: for source mapping only
+        self._stack_frames: NatalStackFrame | None = None
+        if not self.expr:  # expr already has the frame info
+            self._stack_frames = NatalStackFrame()
```

Comment:
I want every component to have a `NatalStackFrame` because it's picked off of the components after the compiler is finished (see line 451 of `pyteal/compiler/compiler.py`). Most of them have `Expr`'s, in which case I'm happy. But in the other case, it makes it easier to avoid special cases downstream.